Sprint_6

eddylasvegas opened the pull request.

Commit:
Обновление README.md и подключи Jacoco, Mockito и JUnit

Commit:
Добавлен .gitignore для файлов IDE

Commit:
Реализованы тесты для Feline, Cat и Lion, добавлены моки и параметризация,Lion не зависит от Feline

Commit:
Отредактирвоан файл README

Commit:
Удалены служебные файлы IDE и сборки, обновлен .gitignore

Commit:
Добавил папку target в репозиторий

Commit:
Добавлена проверка недопустимого пола льва

Commit:
Добавлены: обновлённый .gitignore, отчёты Jacoco

okryukova commented on src/main/java/com/example/Lion.java:
```diff
@@ -5,18 +5,19 @@
+    private final Feline feline;
 
-    public Lion(String sex) throws Exception {
+    public Lion(String sex, Feline feline) throws Exception {
```
✅Отлично. Правильно применен принцип инъекции зависимостей

okryukova commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,31 @@
+    @Test
+    public void testGetSound(){
+        Cat cat = new Cat(feline);
+        assertEquals("Мяу", cat.getSound());
```
⚠️Можно улучшить. Можно использовать поясняющие сообщения для assert. Тогда будет проще разобраться, почему тест упал.

eddylasvegas replied:
Добавлены поясняющие сообщения для assert в тестовых классах

okryukova commented on src/test/java/FelineTest.java:
```diff
@@ -0,0 +1,32 @@
+public class FelineTest {
+    @Test
+    public void testEatMeat() throws Exception{
+        Feline feline = new Feline();
```
⚠️Можно улучшить.  Инициализацию объекта можно вынести в метод с аннотацией before

eddylasvegas replied:
Инициализация объекта вынесена в метод с аннотацией before в тестовых классах CatTest и FelineTest

okryukova commented on src/test/java/LionParametrizedTest.java:
```diff
@@ -0,0 +1,34 @@
+        this.feline = new Feline();
+    }
+
+    @Parameterized.Parameters
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

eddylasvegas replied:
Добавлены именованные параметры тестов к тестовом классе LionParametrizedTest @Parameterized.Parameters(name = "Пол: {0}, Наличие гривы: {1}")

okryukova commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,37 @@
+    @Test
+    public void testGetKittens() throws Exception{
+        Lion lion = new Lion("Самец", feline);
+        when(feline.getKittens()).thenReturn(3);
```
⛔️Нужно исправить.  Мок должен вести себя как реальный объект и возвращать 1

eddylasvegas replied:
Исправлено

Commit:
Рефакторинг тестов:
- Добавлена настройка кодировки UTF-8 в pom.xml
- Добавлены поясняющие сообщения для assert
- Инициализация объектов вынесена в @Before (CatTest, FelineTest)
- Добавлены именованные параметры в LionParametrizedTest
- Исправлен мок Feline в LionTest (getKittens возвращает 1)

Commit:
Обновление отчета JaCoCo после рефакторинга тестов

Commit:
Обновление HTML-отчёта JaCoCo

Commit:
Обновление отчета JaCoCo после рефакторинга тестов 2
